### PR TITLE
feat: add authentication

### PR DESCRIPTION
**Pull request by Bhakti199**

### CODE REVIEW

File: src/Components/NoteEditor/NoteEditor.jsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+            <option>Personal</option>
+            <option>Office</option>
+            <option>Team</option>
+            <option>Project</option>
+          </select>
+          <select className="select-section">
+            <option>Low</option>
+            <option>Medium</option>
+            <option>High</option>
+          </select>
```
Comment (JantuDeb) [inline src/Components/NoteEditor/NoteEditor.jsx:32]: You can create a reusable `<Select>` Component

File: src/Components/RequiresAuth/RequiresAuth.jsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import React from "react";
+import { Navigate, useLocation } from "react-router-dom";
+import { useMainContext } from "../../Context";
+import { LoginPage } from "../../Pages";
+export const RequiresAuth = ({ children }) => {
+  const { isUserLoggedIn } = useMainContext();
+  const location = useLocation();
+  return (
+    <div>
```
Comment (JantuDeb) [inline src/Components/RequiresAuth/RequiresAuth.jsx:9]: `div` is not needed here

**Review by JantuDeb — APPROVED** [review]
Overall the code is structured and clean.

**Comment by tushartiwari7** [comment]
The code seems nice and modular 
Also pretty Ui 🙌

**Review by tushartiwari7 — APPROVED** [review]

**Pull request merged**